webapp/src/containers/Rhs/projectDetails/header.tsx

```diff
@@ -13,27 +13,44 @@ import {toggleIsLinkedProjectListChanged} from 'reducers/linkModal';
 
 import usePluginApi from 'hooks/usePluginApi';
 import useApiRequestCompletionState from 'hooks/useApiRequestCompletionState';
+import IconButton from 'components/buttons/iconButton';
+import SVGWrapper from 'components/svgWrapper';
+import useOutsideClick from 'hooks/useClickOutside';
+import {subscriptionFilterEventTypeReposOptions} from 'plugin_constants/form';
 
 type HeaderProps = {
     projectDetails: ProjectDetails
     showAllSubscriptions: boolean
     setShowAllSubscriptions: (active: boolean) => void
     handlePagination: (reset: boolean) => void
     handleResetProjectDetails: () => void
-    filter: string
-    setFilter: (filter: string) => void
+    filter: SubscriptionFilters
+    setFilter: (filter: SubscriptionFilters) => void
     setSubscriptionList: (subscriptionDetails: SubscriptionDetails[]) => void
 }
 
 const Header = ({projectDetails, showAllSubscriptions, handlePagination, setShowAllSubscriptions, handleResetProjectDetails, filter, setFilter, setSubscriptionList}: HeaderProps) => {
     const {projectName} = projectDetails;
+    const {defaultSubscriptionFilters, subscriptionFilters, filterLabelValuePairAll} = plugin_constants.common;
+    const {subscriptionFilterCreatedByOptions, subscriptionFilterServiceTypeOptions, subscriptionFilterEventTypeBoardsOptions, subscriptionModal} = plugin_constants.form;
     const [showProjectConfirmationModal, setShowProjectConfirmationModal] = useState(false);
 
-    // const [showFilter, setShowFilter] = useState(false); // TODO: uncomment when need to toggle filter
+    const [showFilter, setShowFilter] = useState(false);
 
     const dispatch = useDispatch();
     const {makeApiRequestWithCompletionStatus, getApiState, state} = usePluginApi();
 
+    const getEventTypeOptions = useCallback((serviceType: string) => {
+        switch (serviceType) {
+        case subscriptionFilters.serviceType.boards:
+            return subscriptionFilterEventTypeBoardsOptions();
+        case subscriptionFilters.serviceType.repo:
+            return subscriptionFilterEventTypeReposOptions();
+        default:
+            return [filterLabelValuePairAll];
+        }
+    }, [filter.serviceType]);
```

Comment:
I don't think the function is depending on `filter.serviceType`.

Reply:
yes it is, whenever the `serviceType` changes options for this dropdown changes